### PR TITLE
Tool Interface Enhancements

### PR DESCRIPTION
**Pull request by vlagrassa**
- Render job submission tools from Flask forms
  - Define input validation in Flask forms
- Define species dropdown in Flask forms & use to render on all tool pages
  - Italicize species names in dropdown
  - Prevent user from re-selecting "Choose"
  - Ensure selector is cleared on page load (incl. forward/back button)
- Better handling for disabling form inputs (incl. ARIA)
- Add support email address
- Fix Indel Finder:
  - Query whole chromosome on strain change, even when zoomed in
  - Filter table to only display indels in search range
  - Show error message(s) for invalid start/stop values

### CODE REVIEW

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html [hunk 1]
```diff
@@ -183,7 +167,7 @@ <h1 class="modal-title fs-5" id="generate-primers-modal-label">Thank You</h1>
 		}
 	});
 
-  fetch_strains().then((data) => {
+  fetch_json("{{ url_for('pairwise_indel_finder.get_strains') }}").then((data) => {
```
Comment (r-vieira) [inline src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:170]: Add a `.catch(...)` promise chain to handle possible errors (HTTP network errors, .json parsing errors, etc).

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html [hunk 2]
```diff
@@ -192,11 +176,21 @@ <h1 class="modal-title fs-5" id="generate-primers-modal-label">Thank You</h1>
...
+  // Get form data as an object
+  var data = form_data_to_object('form-submit');
+
+  // Update start and stop to be full chromosome size
+  data['start'] = 1;
+  data['stop']  = current_chrom_size;
+
+  // Run query
   query_running = true;
   $.ajax({
```
Comment (r-vieira) [inline src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:189]: side note: we can standardize network calls using the build-in`fetch()`. Eg: 
```
fetch(url, { method: "POST", body: JSON.stringify{body}, headers: { 'Content-Type': 'application/json'} )
```

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html [hunk 3]
```diff
@@ -138,35 +133,24 @@ <h1 class="modal-title fs-5" id="generate-primers-modal-label">Thank You</h1>
...
+};
+var current_chrom_size = null;
 
 /* Data Table */
 var table;
 
 /* Status */
 var query_running = false
 var loaded = false
+var dragging = false
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html]: Fixed in 021c4889823d376eee014d9123462b7df826d5af

File: src/modules/site-v2/templates/_scripts/utils.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+function fetch_json(url) {
+  return fetch(url)
+    .then( (response) => { return response.text();  } )
+    .then( (text)     => { return JSON.parse(text); } );
+}
+
+
+// Get form data as an object
+function form_data_to_object(form_id) {
+  var data = {};
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/_scripts/utils.js]: Fixed in 021c4889823d376eee014d9123462b7df826d5af

File: src/modules/site-v2/base/views/tools/heritability_calculator.py [hunk 5]
```diff
@@ -102,11 +102,13 @@ def submit_h2():
...
 
-  # TODO: Validate form
+  # Validate form fields
+  # Checks that species is in species list & label is not empty
   if not form.validate_on_submit():
-    pass
-    # flash("You must include a description of your data and a TSV file to upload", "error")
-    # return redirect(url_for('heritability_calculator.submit_h2'))
+    flash("You must include a description of your data and a TSV file to upload.", "danger")
+    return jsonify({
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/base/views/tools/heritability_calculator.py:109]: These return values have been changed significantly in a parallel branch (https://github.com/AndersenLab/CAENDR/tree/enhancement/job-submission-flow), so I'll make this change there.  PR for that branch coming soon.

File: src/modules/site-v2/templates/_scripts/utils.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  // If not, set its background color to mimic a disabled element
+  else {
+    el.style.backgroundColor = `var(--${ val ? 'enabled' : 'disabled' }-color)`;
+  }
+}
+
+
+function fetch_json(url) {
+  return fetch(url)
+    .then( (response) => { return response.text();  } )
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/_scripts/utils.js:28]: Fixed in d439320a0fdf14e797c555f9b5cd70a959261c31

[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/_scripts/utils.js]: Fixed in d439320a0fdf14e797c555f9b5cd70a959261c31

File: src/modules/site-v2/templates/_scripts/utils.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  // If not, set its background color to mimic a disabled element
+  else {
+    el.style.backgroundColor = `var(--${ val ? 'enabled' : 'disabled' }-color)`;
+  }
+}
+
+
+function fetch_json(url) {
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/_scripts/utils.js:26]: Sort of fixed in d439320a0fdf14e797c555f9b5cd70a959261c31.  Not sure the best way to handle failed requests -- should we display to the user, e.g. in a notification bar at the top of the page?  Currently a bit involved to do that in JS alone, esp. without redirecting...

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html (hunk 1 above)
Comment (vlagrassa) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html]: Sort of fixed in d439320a0fdf14e797c555f9b5cd70a959261c31 -- see discussion at https://github.com/AndersenLab/CAENDR/pull/166#discussion_r1154930052.

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html (hunk 2 above)
Comment (vlagrassa) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:193]: By this, do you mean replace the AJAX calls with fetch?  Happy to add that to my backlog if so.

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html [hunk 8]
```diff
@@ -32,23 +31,18 @@
...
-              <div class="mb-4">{{ render_field(form.stop,       disabled='', placeholder="2,029,217") }}</div>
-              <div class="d-grid gap-3 mb-4">
-                <button id='find_indels' type='submit' value="submit" class="btn btn-secondary text-light" disabled>Find Indels</button>
-              </div>
-              </form>
-              <!-- /Input Form -->
-              {% include "_includes/ignore-cache-checkbox.html" %}
-              </div>
-              <!-- /Input Card -->
+              <div class="mb-4">{{ render_field(form.start,      disabled='', placeholder="2,028,824", make_err_container=true) }}</div>
```
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:38]: Now that you point it out, I think we could actually remove these entirely.  They're auto-filled when the user selects a chromosome, and the user is prevented from entering invalid values.

File: src/modules/site-v2/templates/_scripts/utils.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (r-vieira) [reply src/modules/site-v2/templates/_scripts/utils.js:26]: Good idea -- a notification to the end user works. (Ask the user to try again later or contact support if it becomes a recurring issue.)

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html (hunk 8 above)
[the remark this answers is not in the document]
Comment (r-vieira) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:38]: Even better! Thanks

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html (hunk 2 above)
Comment (r-vieira) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:193]: Yes, please -- this is part of the master-plan to remove dependencies on the jQuery lib.

File: src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html (hunk 8 above)
[the remark this answers is not in the document]
Comment (vlagrassa) [reply src/modules/site-v2/templates/tools/pairwise_indel_finder/submit.html:38]: Update -- started a quick to-do thread about this on Basecamp.

**Pull request merged**